### DISCUSSION TITLE
Allow us to name the Block

### DISCUSSION THREAD

**Discussion by prjseal** [umbraco/Umbraco-CMS#11749] [Features and ideas]
I've been creating a lot of blocks lately. One of the things I have noticed which would be really helpful is the ability to name the block ourselves.

I'm not talking about the label, I know I can edit that already. I'm talking about the name of the block as it appears in the available blocks, and what the editor sees when choosing a block to add.

![image](https://user-images.githubusercontent.com/9142936/145418648-14da9365-0102-48d3-a2ed-b0ebd097a16e.png)

Currently the name of the block comes from the name of the document type. But I would like to use the same document type on a different block in a different context, so it would be great if I could name that block differently.

See this image:

![image](https://user-images.githubusercontent.com/9142936/145417816-0264875a-be67-4be3-ad22-fe56b8a114d7.png)

I would like to name the block 'Spotlight Item'. The fact that it shares the Carousel Item document type shouldn't matter. I should be able to use it for a different purpose and give it a different name without needing to create a new document type.

So just to recap. My request is simple. Please give us the ability to override the name of the block. Perhaps if it is left empty it defaults to the name of the content model document type.

**Comment by c9mb** [comment]
Similar to naming Nested Items ?

**Comment by mzajkowski** [comment]
Nice idea! I had a WIP PR for Nested Content items a while ago and I found it extremely useful in many cases. +1 then!
